File: week17/week17_2.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        N = len(digits) #有幾位數
        carry = 1
        for i in range(N-1, -1, -1): #倒過來的迴圈
            digits[i] = digits[i] + carry #加1
            if digits[i] >= 10: #如果加完 >= 10 有進位

                carry = 1 #進位
                digits[i] = digits[i] % 10 #餘數, 留在原地
            else: #沒有進位
                carry = 0 #就設成0
        if carry==0:
            return digits
        else:
            return [carry] + digits

File: week17/week17_2.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
       int N = digits.size();
       int carry = 1;
       for(int i=N-1; i>=0; i--) {
            digits[i] = digits[i] + carry;
            if(digits[i]>=10) { //大於等於, 進位
                carry = 1;
                digits[i] = digits[i] % 10; //餘數
            } else {
                carry = 0;
            }
       }

       if(carry==0) return digits; //沒有進位, 直接當答案
       else {
            vector<int> ans(N+1); //多1位(進位的carry)
            ans[0] = carry;
            for(int i=0; i<N; i++) {
                ans[i+1] = digits[i]; //左邊要避開 carry 的那一位
            }
            return ans;
       }
        
    }
};